### PR TITLE
refactor: split get_submission_and_assessment_info into two handlers

### PR DESCRIPTION
**Pull request by jansenk**
**TL;DR -** split `get_submission_and_assessment_info` into two handlers, `get_submission_info` and `get_assessment_info`

JIRA: [AU-365](https://openedx.atlassian.net/browse/AU-365) (this PR isn't directly for this, but it enables and simplifies it)
 - I also split the tests, and pulled shared testing to `test_base.py`

**Developer Checklist**

- [ ] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Translations and JS/SASS compiled
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

File: openassessment/staffgrader/tests/test_get_assessment_info.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+            'feedback': "Base Assessment Feedback",
+            'score': {
+                'pointsEarned': 0,
+                'pointsPossible': 0,
+            },
+            'criteria': [
+                OrderedDict({
+                    'name': part.criterion.name,
+                    'option': None,
+                    'points': None,
```
Comment (jansenk) [inline openassessment/staffgrader/tests/test_get_assessment_info.py:250]: I made sure to not lose these changes that I literally just merged in lol

**Review by leangseu-edx — APPROVED** [review]
LGTM

**Review by nsprenkle — APPROVED** [review]
Looks good
